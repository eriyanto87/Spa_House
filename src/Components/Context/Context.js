import React from "react";

const Context = React.createContext({
  // users: [],
  // cart: [],
  // treatments: [],
  // addUser: () => {},
  // addCart: () => {},
  //   getAllTreatments: () => {},
});

export default Context;
