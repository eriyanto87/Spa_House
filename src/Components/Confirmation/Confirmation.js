import React from "react";
import { Link } from "react-router-dom";
import Context from "../Context/Context";

export default function Confirmation() {
  return (
    <Context.Consumer>
      {(context) => {
        console.log(context);
        const latestUserIndex = context.users.length - 1;
        console.log(context.users[latestUserIndex].user_name, "name");
        return (
          <div>
            <section>
              <h3> {context.users[latestUserIndex].user_name}, </h3>
              <p>
                Your appoinment is set at {context.displayTime} on{" "}
                {context.displayDate}
              </p>
              {/* <h3>How To Prep For Your Appointment</h3>
              <h4>Massage</h4>
              <h4>Lash</h4>
              <h4>Nails</h4>
              <h4>Skin</h4> */}
              <Link to="/">
                <h3>Home</h3>
              </Link>
            </section>
          </div>
        );
      }}
    </Context.Consumer>
  );
}
