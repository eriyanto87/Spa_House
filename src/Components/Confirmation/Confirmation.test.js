import React from "react";
import ReactDOM from "react-dom";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import Context from "../Context/Context";
import Confirmation from "./Confirmation";

describe("Confirmation component", () => {
  const contextValues = {
    users: [
      {
        user_name: "",
      },
    ],
    displayTime: "",
    displayDate: "",
  };

  it("Confirmation Snapshot", () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Context.Provider value={contextValues}>
            <Confirmation />
          </Context.Provider>
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <BrowserRouter>
        <Context.Provider value={contextValues}>
          <Confirmation />
        </Context.Provider>
      </BrowserRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });
});
