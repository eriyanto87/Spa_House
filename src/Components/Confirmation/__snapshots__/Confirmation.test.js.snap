// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirmation component Confirmation Snapshot 1`] = `
<div>
  <div
    className="nav"
  >
    <ul>
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          Home 
        </a>
      </li>
      <li>
        <a
          href="/about"
          onClick={[Function]}
        >
          About 
        </a>
      </li>
      <li>
        <a
          href="/treatments"
          onClick={[Function]}
        >
          Treatments 
        </a>
      </li>
      <li>
        <a
          href="/pricing"
          onClick={[Function]}
        >
          Pricing 
        </a>
      </li>
      <li>
        <a
          href="/contact"
          onClick={[Function]}
        >
          Contact Us 
        </a>
      </li>
      <li>
        <a
          href="/gallery"
          onClick={[Function]}
        >
          Gallery
        </a>
      </li>
    </ul>
  </div>
  <section>
    <h3>
       
      
      , 
    </h3>
    <p>
      Your appoinment is set at 
      
       on
       
      
    </p>
  </section>
  <div>
    <p>
      Â© 2020
    </p>
  </div>
</div>
`;
